crates/models/src/journals.rs

```diff
@@ -21,8 +21,14 @@ pub struct BucketAndPrefix {
 }
 
 impl BucketAndPrefix {
-    fn bucket_and_prefix(&self) -> (&str, &str) {
-        (self.bucket.as_str(), self.prefix.as_deref().unwrap_or(""))
+    fn as_url(&self, scheme: &str) -> url::Url {
+        url::Url::parse(&format!(
+            "{}://{}/{}",
+            scheme,
+            self.bucket,
+            self.prefix.as_deref().unwrap_or("")
+        ))
+        .expect("parsing as URL should never fail")
```

Comment:
I don't think the `bucket` is actually guaranteed to be valid at the point when this is called. If the bucket validates against `BUCKET_RE`, then parsing should never fail. But it seems that our [validation logic](https://github.com/estuary/flow/blob/db2cdd86825132ee7e0bcac8b432712ab5866c83/crates/validation/src/storage_mapping.rs#L5) never actually validates the bucket name at all. Can you update that to add a validation?

IMO there ought to also be a comment here saying that the function may panic if the bucket and prefix have not been validated.

Reply:
Good point, done!